_posts/2020-05-13-immutable-keyword.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.6.x features: Saving Storage Costs with Immutables'
+date: '2020-05-13'
+author: Daniel Kirchner
+category:
+---
+
+With [version 0.6.5](https://github.com/ethereum/solidity/releases/tag/v0.6.5), Solidity introduced the 
+[``immutable`` keyword](https://solidity.readthedocs.io/en/v0.6.7/contracts.html?highlight=immutable#constant-and-immutable-state-variables) 
+for state variables. Immutable state variables can only be assigned during contract creation, but will remain 
+constant throughout the life-time of a deployed contract.
+The big advantage of immutables is that reading them is significantly cheaper than reading from regular state variables, 
+since immutables will not be stored in storage, but their values will be directly inserted into the runtime code.
+
+## How to use Immutables
+
+Immutable state variables can be declared using the ``immutable`` keyword. They cannot be read during contract creation, 
+but either have to be written to exactly once in the constructor or initialized directly in their declaration. 
+Runtime code can only read immutables, but not write to them.
+<br><br>
+Here is an example of ``immutable`` in action:
+```
```

Comment:
```suggestion
```solidity
```

Reply:
(not sure if it works already)

Reply:
I think it doesn't or was there an update on the blog config? See https://github.com/ethereum/solidity-blog/issues/34

Reply:
Actually let me try - if this does not break it, we do not have to add it later on.